src/videodownlinkbandwidthpolicy/VideoAdaptiveProbePolicy.ts

```diff
@@ -3,15 +3,30 @@
 
 import ContentShareConstants from '../contentsharecontroller/ContentShareConstants';
 import Logger from '../logger/Logger';
+import ServerSideNetworkAdaption from '../signalingclient/ServerSideNetworkAdaption';
 import VideoStreamDescription from '../videostreamindex/VideoStreamDescription';
 import VideoStreamIndex from '../videostreamindex/VideoStreamIndex';
 import VideoPreference from './VideoPreference';
 import { VideoPreferences } from './VideoPreferences';
 import VideoPriorityBasedPolicy from './VideoPriorityBasedPolicy';
+import VideoPriorityBasedPolicyConfig from './VideoPriorityBasedPolicyConfig';
 
+/** [[VideoAdaptiveProbePolicy]] wraps [[VideoPriorityBasedPolicy]] with customized behavior to automatically
+ * assign a high preference to content share.
+ *
+ * @deprecated This class is not compatible with latest priority policy features (i.e. server side network adaptation),
+ * [[VideoPriorityBasedPolicy]] should be used instead. If high priority for content is desired it can be done trivially
+ * by the application.
+ */
 export default class VideoAdaptiveProbePolicy extends VideoPriorityBasedPolicy {
+  private static createConfig(): VideoPriorityBasedPolicyConfig {
```

Comment:
Hmm why static? Do we need to make this singleton for multiple `VideoAdaptiveProbePolicy` objects? If so checking if this approach is useful or do we need explicit object checking to preserve a singleton use of the config.

Reply:
I'll add a comment. This is just a workaround for the silly typescript rule that `super` calls must be first in constructor. We don't need singleton usage (even though i'm not sure what customer would be creating multiple policies lol).

Reply:
Actually i misunderstood comments i wrote before. We don't need to deprecate this class, my bad.